bin-src/lib/getDependentStoryFiles.ts

```diff
@@ -31,6 +21,55 @@ const isUserModule = (mod: Module | Reason) =>
 // already contains forward slashes, because that's what git yields even on Windows.
 const posix = (localPath: string) => localPath.split(path.sep).filter(Boolean).join(path.posix.sep);
 
+// put this in file with same name
+const getExternalChangedDirectDependencies = async (changedFiles) => {
+  const dependenciesChanged = [];
+  // const changedPackageFiles = changedFiles.filter(changedFile => /(^package\.json$)|(\/package\.json$)/.test(changedFile))
+  const changedPackageFiles = ['package.json'];
+  try {
+    await Promise.all(
+      changedPackageFiles.map(async (packageFilePath) => {
+        // do same build trick to get old commit
+        const changes = await execGitCommand(`git --no-pager diff ${packageFilePath}`);
+        const filez = parseDiff(changes);
+        filez[0].chunks.forEach((chunk) => {
+          const theChanges = chunk.changes.filter((item) => item.type === 'add');
+          // make sure this change is a proper change... should be easy enough...
+          // honestly just pull out the package name
+          // (later check that it is in proper format, maybe just using semver)
+          const refinedChanges = theChanges.map((aChange) =>
+            aChange.content.split('+')[1].split(':')[0].trim().replace(/"/g, '')
+          );
+
+          const jsonFile = fs.readFileSync(packageFilePath, 'utf-8');
+          const parsedJsonFile = JSON.parse(jsonFile);
+          // get it out of there, if it exists...
+          refinedChanges.forEach((change) => {
+            const key = change;
+            const value =
+              parsedJsonFile.dependencies?.[key] ||
+              parsedJsonFile.devDependencies?.[key] ||
+              parsedJsonFile.peerDependencies?.[key];
+
+            // if this value exists there, we do want to mark things as needing changes
+            if (value) {
+              dependenciesChanged.push(key);
+            } else {
+              console.log('not this changed...');
+            }
+          });
+        });
+      })
+    );
+
+    console.log('changed', dependenciesChanged);
+  } catch (err) {
+    console.error(err);
+  }
+
+  return dependenciesChanged;
+};
```

Comment:
This function needs to move to its own dedicated file, and I need to finesse the string-splitting and other stuff. Also, for now it's actually using the unstaged package.json's changes for the diff, instead of diffing based off of the baseline's commit. Will work on that next week.